### PR TITLE
chore(compass-editor): add copy and format buttons to multiline ace editors

### PR DESCRIPTION
**Pull request by gribnoysup**
Visible by default on focus / hover for all editors, but tell me if you think we should hide it somewhere

### CODE REVIEW

File: packages/compass-aggregations/src/components/pipeline-builder-workspace/pipeline-as-text-workspace/pipeline-editor.tsx [hunk 1]
```diff
@@ -34,15 +34,14 @@ const containerStyles = css({
...
-  overflow: 'auto',
+  overflow: 'hidden',
 });
 
 const errorContainerStyles = css({
   flex: 'none',
   marginTop: 'auto',
   marginLeft: spacing[3],
   marginRight: spacing[3],
-  marginBottom: spacing[2],
```
Comment (gribnoysup) [inline packages/compass-aggregations/src/components/pipeline-builder-workspace/pipeline-as-text-workspace/pipeline-editor.tsx:45]: @mabaasit any reason we returned this bottom margin and the one below? I removed them in a PR about a week ago because they make bottom content misaligned (and make it hard to move content around in the editor because they clash with the paddings of the container):

image
image

File: packages/compass-editor/src/ace/theme.ts [hunk 2]
```diff
@@ -136,6 +138,23 @@ const mongodbAceThemeCssText = css`
   .ace_dark.ace-mongodb .ace_gutter-cell.ace_error {
     background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='none' viewBox='0 0 16 16'%3E%3Cpath fill='${errorColorDarkMode}' fill-rule='evenodd' d='M8 15A7 7 0 1 0 8 1a7 7 0 0 0 0 14zm1.414-9.828a1 1 0 1 1 1.414 1.414L9.414 8l1.414 1.414a1 1 0 1 1-1.414 1.414L8 9.414l-1.414 1.414a1 1 0 1 1-1.414-1.414L6.586 8 5.172 6.586a1 1 0 0 1 1.414-1.414L8 6.586l1.414-1.414z' clip-rule='evenodd'/%3E%3C/svg%3E%0A");
   }
+  .ace-mongodb .ace_fold-widget {
+    /* leafygreen CaretDown */
+    background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='none' viewBox='0 0 16 16'%3E%3Cpath fill='${foldColor}' d='M8.679 10.796a.554.554 0 0 1-.858 0L4.64 6.976C4.32 6.594 4.582 6 5.069 6h6.362c.487 0 .748.594.43.976l-3.182 3.82z'/%3E%3C/svg%3E%0A");
```
Comment (gribnoysup) [inline packages/compass-editor/src/ace/theme.ts:143]: drive-by: replacing another raster icon with an svg version from leafygreen

File: packages/compass-editor/src/multiline-editor.tsx [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+};
+
+type EditorProps = Omit<BaseEditorProps, 'editorClassName'> & {
+  copyable?: boolean;
+  formattable?: boolean;
+  customActions?: CustomActionItem[];
+  onAction?: (editor: Ace.Editor | undefined, action: string) => void;
+};
+
+const FormatIcon = ({
```
Comment (gribnoysup) [inline packages/compass-editor/src/multiline-editor.tsx:23]: Copied from figma, there is no icon like that in leafygreen

File: packages/compass-aggregations/src/components/pipeline-builder-workspace/pipeline-as-text-workspace/pipeline-editor.tsx (hunk 1 above)
Comment (mabaasit) [reply packages/compass-aggregations/src/components/pipeline-builder-workspace/pipeline-as-text-workspace/pipeline-editor.tsx:45]: it was a mistake to do that. the changes you've made in this PR regarding spacing fixes it.

File: packages/compass-editor/src/multiline-editor.tsx [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+  const actions = useMemo(() => {
+    return [
+      copyable && (
+        <Button
+          size="xsmall"
+          aria-label="Copy"
+          title="Copy"
+          onClick={() => {
+            editorRef.current?.execCommand('copy-all');
```
Comment (mabaasit) [inline packages/compass-editor/src/multiline-editor.tsx:90]: Should we show some kind of feedback once copied?

File: packages/compass-aggregations/src/components/pipeline-builder-workspace/pipeline-as-text-workspace/pipeline-editor.tsx (hunk 1 above)
Comment (gribnoysup) [reply packages/compass-aggregations/src/components/pipeline-builder-workspace/pipeline-as-text-workspace/pipeline-editor.tsx:45]: Cool, thanks! Just double-checking I am not missing some corner-case state that this was accounting for

File: packages/compass-editor/src/multiline-editor.tsx (hunk 4 above)
Comment (gribnoysup) [reply packages/compass-editor/src/multiline-editor.tsx:90]: Hmm, seems like leafygreen copy button flashes checkmark icon for a second to indicate that. Yeah, probably worth doing this both for copy and format, nice suggestion!
